### ISSUE TITLE
Show-HPOVFirmwareReport Baseline Version N/A

### ISSUE THREAD

**Issue by lga0101** [HewlettPackard/POSH-HPOneView#374]
Please fill in as much information as possible to help resolve your issue.

### Expected Behavior
Firmware comparison between Server and Baseline components

### Actual Behavior
Generates a server current FW levels , but the provided baseline version is reported as N/A. No comparison between them.
 
Not sure if that is the current expected behavior or the issue has been already reported 

##########
 Show-HPOVFirmwareReport -InputObject $server -Baseline $Baseline -Export
#########

```PowerShell
VERBOSE: [SEND-HPOVREQUEST] Manual Pagination: False
VERBOSE: [SEND-HPOVREQUEST] Response object, no paging needed.
VERBOSE: [SEND-HPOVREQUEST] Cleaning up HttpWebRequest
VERBOSE: [SEND-HPOVREQUEST] Does nextPageUri member exist: False
VERBOSE: [SEND-HPOVREQUEST] Is nextPageUri Null or Empty: True
VERBOSE: [SEND-HPOVREQUEST] Stopping Do/Until loop because nextPageUri doesn't exist and have received all objects.
VERBOSE: [SEND-HPOVREQUEST] End
VERBOSE: [GET-SERVERFIRMWARE] Server Hardware Type supports firmware management.
VERBOSE: [GET-SERVERFIRMWARE] Baseline value provided: 

Name        : Service Pack for ProLiant
Version     : 2018.9.0
ISOFileName : P09835_001_spp-2018_09_0-SPP2018090_2018_0914_32.iso
XMLKeyName  : bp003318
BundleSize  : 6.02GB
Locations   : {Internal}




VERBOSE: [GET-SERVERFIRMWARE] Baseline resource passed.
VERBOSE: [GET-SERVERFIRMWARE] Baseline resource name: SPP 2018.09.0
VERBOSE: [GET-SERVERFIRMWARE] Baseline resource uri: /rest/firmware-drivers/P09835_001_spp-2018_09_0-SPP2018090_2018_0914_32
VERBOSE: [SEND-HPOVREQUEST] BEGIN
VERBOSE: [SEND-HPOVREQUEST] Called from: Get-ServerFirmware
VERBOSE: [SEND-HPOVREQUEST] Bound PS Parameters: 
Key      Value                                    
---      -----                                    
uri      /rest/server-hardware/firmware-compliance
method   POST                                     
body     {serverUUID, firmwareBaselineId}         
Hostname oneview                                  



VERBOSE: [SEND-HPOVREQUEST] Process
VERBOSE: [SEND-HPOVREQUEST] Hostname value: oneview

VERBOSE: [SEND-HPOVREQUEST] Filtering for Connection Object via String: oneview
VERBOSE: [SEND-HPOVREQUEST] Processing 'oneview' appliance connection request. 1 of 1
VERBOSE: [SEND-HPOVREQUEST] Requested URI '/rest/server-hardware/firmware-compliance' to 'oneview'
VERBOSE: [RESTCLIENT] Building new [System.Net.HttpWebRequest] object for POST https://oneview/rest/server-hardware/firmware-compliance
VERBOSE: [SEND-HPOVREQUEST] Body object found. Converting to JSON.
VERBOSE: [SEND-HPOVREQUEST] HTTP Method is POST. Removing 'ApplianceConnection' NoteProperty from object(s).
VERBOSE: [SEND-HPOVREQUEST] Request Body: {"serverUUID":"39373638-3935-584D-5138-313630394730","firmwareBaselineId":"P09835_001_spp-2018_09_0-S
PP2018090_2018_0914_32"}
VERBOSE: [SEND-HPOVREQUEST] Request: POST https://oneview/rest/server-hardware/firmware-compliance
VERBOSE: [SEND-HPOVREQUEST] Request Header 1: User-Agent = HPEOneview_PowerShell/4.10 (Microsoft Windows NT 6.3.9600.0)
VERBOSE: [SEND-HPOVREQUEST] Request Header 2: Content-Type = application/json
VERBOSE: [SEND-HPOVREQUEST] Request Header 3: Accept = application/json, text/html, application/xhtml+xml
VERBOSE: [SEND-HPOVREQUEST] Request Header 4: X-API-Version = 800
VERBOSE: [SEND-HPOVREQUEST] Request Header 5: accept-language = en_US
VERBOSE: [SEND-HPOVREQUEST] Request Header 6: accept-encoding = gzip, deflate
VERBOSE: [SEND-HPOVREQUEST] Request Header 7: auth = [*****REDACTED******]
VERBOSE: [SEND-HPOVREQUEST] Request Header 8: Host = oneview
VERBOSE: [SEND-HPOVREQUEST] Response time: 00:00:00.1155800
VERBOSE: [SEND-HPOVREQUEST] Response Status: 200 (OK)
VERBOSE: [SEND-HPOVREQUEST] Response Header 1: Transfer-Encoding = chunked
VERBOSE: [SEND-HPOVREQUEST] Response Header 2: Cache-Control = no-cache
VERBOSE: [SEND-HPOVREQUEST] Response Header 3: Content-Type = application/json;charset=UTF-8
VERBOSE: [SEND-HPOVREQUEST] Response Header 4: Date = Mon, 29 Oct 2018 17:34:33 GMT
VERBOSE: [SEND-HPOVREQUEST] Response Header 5: Server = Apache
VERBOSE: [SEND-HPOVREQUEST] FinalResponse: {"componentMappingList":[{"componentKey":null,"componentLocation":"System Board","componentName":"In
telligent Platform Abstraction Data","componentType":"Firmware","installedVersion":"6.1.0 Build 12","baselineVersion":"unknown","componentFirmw
areUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":" ","componentName":"nvme drive eject nmi fix versionin
g dll","componentType":"Software","installedVersion":"1.1.0.0","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManage
d":false},{"componentKey":null,"componentLocation":"System Board","componentName":"Server Platform Services (SPS) Firmware","componentType":"Fi
rmware","installedVersion":"4.0.4.288","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey
":null,"componentLocation":"System Board","componentName":"Intelligent Provisioning","componentType":"Firmware","installedVersion":"3.10.222","
baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":"System Board
","componentName":"Innovation Engine (IE) Firmware","componentType":"Firmware","installedVersion":"0.1.5.2","baselineVersion":"unknown","compon
entFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":" ","componentName":"hp lights-out online confi
guration utility","componentType":"Software","installedVersion":"5.2.0.0","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"
hpsumManaged":false},{"componentKey":null,"componentLocation":"System Board","componentName":"Power Management Controller FW Bootloader","compo
nentType":"Firmware","installedVersion":"1.1","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"compo
nentKey":null,"componentLocation":" ","componentName":"emulex storport miniport driver","componentType":"Software","installedVersion":"11.4.334
.7","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":"a6b1a447-382a-5a4f-14e4-1657103c
22be","componentLocation":"Embedded LOM","componentName":"HPE Ethernet 1Gb 4-port 331i Adapter - NIC","componentType":"Firmware","installedVers
ion":"20.12.41","baselineVersion":"20.12.41","componentFirmwareUpdateRequired":false,"hpsumManaged":true},{"componentKey":null,"componentLocati
on":"Bay 2","componentName":"Power Supply Firmware","componentType":"Firmware","installedVersion":"1.00","baselineVersion":"unknown","component
FirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":"Bay 1","componentName":"Power Supply Firmware","c
omponentType":"Firmware","installedVersion":"1.00","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"
componentKey":null,"componentLocation":" ","componentName":"integrated smart update tools","componentType":"Software","installedVersion":"2.3.0
.0","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":" ","com
ponentName":"smart storage administrator","componentType":"Software","installedVersion":"3.30.14.0","baselineVersion":"unknown","componentFirmw
areUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":"Embedded Device","componentName":"HPE Smart Storage Ba
ttery 1 Firmware","componentType":"Firmware","installedVersion":"0.60","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hps
umManaged":false},{"componentKey":"00000000-0000-0000-0000-000000000204","componentLocation":"System Board","componentName":"System ROM","compo
nentType":"Firmware","installedVersion":"U32 v1.42 (06/20/2018)","baselineVersion":"06/20/2018","componentFirmwareUpdateRequired":false,"hpsumM
anaged":true},{"componentKey":"4764a662-b342-4fc7-9ce9-258c5d99e815","componentLocation":"System Board","componentName":"iLO 5","componentType"
:"Firmware","installedVersion":"1.35 Aug 14 2018","baselineVersion":"1.35","componentFirmwareUpdateRequired":false,"hpsumManaged":true},{"compo
nentKey":null,"componentLocation":"Embedded Device","componentName":"Embedded Video Controller","componentType":"Firmware","installedVersion":"
2.5","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":"a6b1a447-382a-5a4f-15b3-1015159
000d3","componentLocation":"Embedded ALOM","componentName":"HPE Eth 10/25Gb 2p 640FLR-SFP28 Adptr","componentType":"Firmware","installedVersion
":"14.22.14.14","baselineVersion":"14.22.1414","componentFirmwareUpdateRequired":true,"hpsumManaged":true},{"componentKey":null,"componentLocat
ion":" ","componentName":"hp proliant wmi providers for windows server 2003/2008/2012","componentType":"Software","installedVersion":"10.60.0.0
","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":" ","compo
nentName":"csver","componentType":"Software","installedVersion":"10.1.2.86","baselineVersion":"unknown","componentFirmwareUpdateRequired":false
,"hpsumManaged":false},{"componentKey":"a6b1a447-382a-5a4f-10df-e30015900213","componentLocation":"PCI-E Slot 1","componentName":"HPE SN1200E 1
6Gb 1p FC HBA","componentType":"Firmware","installedVersion":"11.4.334.10","baselineVersion":"11.4.334.31","componentFirmwareUpdateRequired":tr
ue,"hpsumManaged":true},{"componentKey":null,"componentLocation":" ","componentName":"ilo 5 channel interface driver","componentType":"Software
","installedVersion":"4.3.0.0","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"
componentLocation":" ","componentName":"broadcom netxtreme gigabit ethernet ndis6.x unified driver.","componentType":"Software","installedVersi
on":"212.0.0.0","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocati
on":"Embedded controller Port 1I Box 1 Bay 2","componentName":"Drive","componentType":"Firmware","installedVersion":"HPG6","baselineVersion":"u
nknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":"System Board","componentName":"
System Programmable Logic Device","componentType":"Firmware","installedVersion":"0x2A","baselineVersion":"unknown","componentFirmwareUpdateRequ
ired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":"Embedded controller Port 1I Box 1 Bay 1","componentName":"Drive","c
omponentType":"Firmware","installedVersion":"HPG6","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"
componentKey":null,"componentLocation":" ","componentName":"agentless management service","componentType":"Software","installedVersion":"1.30.0
.0","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":" ","com
ponentName":"ilo 5 asr driver","componentType":"Software","installedVersion":"4.2.0.0","baselineVersion":"unknown","componentFirmwareUpdateRequ
ired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":" ","componentName":"Matrox G200eH Display Only Driver","componentTy
pe":"Software","installedVersion":"9.15.1.184","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"comp
onentKey":null,"componentLocation":" ","componentName":"smart storage administrator command line interface","componentType":"Software","install
edVersion":"3.30.14.0","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componen
tLocation":" ","componentName":"hpe insight management agents version dll","componentType":"Software","installedVersion":"10.60.0.0","baselineV
ersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLocation":"System Board","compon
entName":"Redundant System ROM","componentType":"Firmware","installedVersion":"U32 v1.36 (02/14/2018)","baselineVersion":"unknown","componentFi
rmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":"00000000-0000-0000-0000-000000504d05","componentLocation":"System Board","co
mponentName":"Power Management Controller Firmware","componentType":"Firmware","installedVersion":"1.0.4","baselineVersion":"1.0.4","componentF
irmwareUpdateRequired":false,"hpsumManaged":true},{"componentKey":null,"componentLocation":" ","componentName":"smartraid, smarthba pqi storpor
t driver","componentType":"Software","installedVersion":"100.62.0.64","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsu
mManaged":false},{"componentKey":"a6b1a447-382a-5a4f-9005-028f103c0602","componentLocation":"Embedded RAID","componentName":"HPE Smart Array P4
08i-a SR Gen10","componentType":"Firmware","installedVersion":"1.65","baselineVersion":"1.65","componentFirmwareUpdateRequired":false,"hpsumMan
aged":true},{"componentKey":null,"componentLocation":" ","componentName":"smart storage administrator","componentType":"Software","installedVer
sion":"3.30.14.0","baselineVersion":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false},{"componentKey":null,"componentLoca
tion":" ","componentName":"combined chipset identifier versioning dll","componentType":"Software","installedVersion":"8.2.0.0","baselineVersion
":"unknown","componentFirmwareUpdateRequired":false,"hpsumManaged":false}],"serverFirmwareUpdateRequired":true}
VERBOSE: [SEND-HPOVREQUEST] Manual Pagination: False
VERBOSE: [SEND-HPOVREQUEST] Response object, no paging needed.
VERBOSE: [SEND-HPOVREQUEST] Cleaning up HttpWebRequest
VERBOSE: [SEND-HPOVREQUEST] Does nextPageUri member exist: False
VERBOSE: [SEND-HPOVREQUEST] Is nextPageUri Null or Empty: True
VERBOSE: [SEND-HPOVREQUEST] Stopping Do/Until loop because nextPageUri doesn't exist and have received all objects.
VERBOSE: [SEND-HPOVREQUEST] End
VERBOSE: [SHOW-HPOVFIRMWAREREPORT] Completed Collecting Server Firmware Information - Skipping Write-Progress display.

### Version Information
oneview                             LibraryVersion                                                                    
-------                             -------------- ----                                                                   
ApplianceVersion: 4.10.01.348545.00 4.10.1821.1567 


$PSVersionTable
Name                           Value                                                                                                          
----                           -----                                                                                                          
PSVersion                      5.1.14409.1005                                                                                                 
PSEdition                      Desktop                                                                                                        
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                        
BuildVersion                   10.0.14409.1005                                                                                                
CLRVersion                     4.0.30319.42000                                                                                                
WSManStackVersion              3.0                                                                                                            
PSRemotingProtocolVersion      2.3                                                                                                            
SerializationVersion           1.1.0.1  
```PowerShell

```

**Comment by ChrisLynchHPE** [comment]
Can you share the report?  What server hardware and generation are you trying to run this report for?

**Comment by lga0101** [comment]
Report is attached, just removed the "Name" column .
Server: ProLiant DL360 Gen10
SPP: 2018.09

Also tested on ProLiant DL380 Gen9.
Thank you.

На ср, 31.10.2018 г. в 12:46 ч. Chris Lynch <notifications@github.com>
написа:

**Comment by lga0101** [comment]
[Report.zip](https://github.com/HewlettPackard/POSH-HPOneView/files/2538689/Report.zip)

**Comment by ChrisLynchHPE** [comment]
Thanks.  Can you use `Get-HPOVCommandTrace` to capture a command trace:

```PowerShell
Get-HPOVCommandTrace -ScriptBlock { Show-HPOVFirmwareReport -InputObject $server -Baseline $Baseline -Export }
```

And then provide me the captured file?

**Comment by lga0101** [comment]
Changes User and Server names. Thank you. 

[1811021403_HPOV_transcript.log](https://github.com/HewlettPackard/POSH-HPOneView/files/2543638/1811021403_HPOV_transcript.log)

**Comment by ChrisLynchHPE** [comment]
Thanks.  I see the issue, and will have this fixed in the next library update.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 4.10.1889.2173](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v4.10.1889.2173) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.410/4.10.1889.2173).

Please verify before this issue is closed.

**Comment by ChrisLynchHPE** [comment]
Closing do to no further activity.
